src/XrdVoms/XrdVomsHttp.cc

```diff
@@ -142,7 +143,9 @@ XrdHttpSecXtractor *XrdHttpGetSecXtractor(XrdHttpSecXtractorArgs)
 
 // Now return the interface object
 //
-   return (XrdHttpSecXtractor *)new XrdVomsHttp(eDest, *vomsFun);
+   auto base = static_cast<XrdHttpSecXtractor *>(new XrdVomsHttp(eDest, *vomsFun));
+   auto wrapper = static_cast<XrdHttpSecXtractor *>(XrdVomsMapfile::Configure(eDest, base));
+   return wrapper ? wrapper : base;
```

Comment:
I have a problem with this a) it totally ignores any errors that may occur during configuration and this has always been a site complaint about silent errors that should be fatal, and b) this could have been done in a much more straightforward way by simply inserting a test for mapping in XrdVomsHttp::GetSecData() instead using an indirect wrapper. Though, frankly, I would do it the other way around and pass a mapper to the voms function itself (or a null pointer) so that this works for http as well as xrootd. Then we have a chance to make it consistent across all protocols.

Reply:
I did the wrapper because I thought that's what we agreed to in the phone call -- keep the code outside the existing XrdVoms class.  Will be happy to put it inside the `XrdVomsHttp` class itself if you're OK with that.

I am reluctant to create a new version of the plugin protocol as that would probably require me to add even more arguments to the GSI configuration, which is already much too long.

Reply:
Two comments:
1) Indeed we did agree to that but that was before issue 1577 which requested we converge the voms mapfile support. That is a good idea but for a clean convergence the code would have to be moved into the plugin. However, that's fairly easy to do given the wrapper which is used for xroot and http. So, that's why I did a turn-around. 

2) Actually, you really don't have to change the plugin protocol, just enhance it. So, if you return MAP_FAILED instead of zero then the caller will know there was a fatal error while zero (still used by old plugins) would mean to ignore the plugin for whatever reason. Note that MAP_FAILED is defined in sys/mman.h and specifically is used to indicate that a memory address could not be established due to am error.

Reply:
Latest commit refactors this class to be a core part of the plugin as requested.  Resolving.